Content.Shared/Atmos/Monitor/AtmosAlarmThresholdPrototype.cs

```diff
@@ -0,0 +1,215 @@
+        private float? _upperWarningPercentage;
+        [ViewVariables]
+        [DataField("upperWarnAround")]
+        public float? UpperWarningPercentage
+        {
+            get => _upperWarningPercentage;
+            set
+            {
+                if (value == null)
+                {
+                    _upperWarningPercentage = null;
+                    return;
+                }
+
+                var testValue = value * UpperBound;
+                if (value > 1f
+                    || testValue < LowerWarningBound
+                    || testValue < LowerBound)
+                    return;
+
+                _upperWarningPercentage = value;
+            }
+
+        }
+
+        // lower warning percentage
+        // must always cause LowerWarningBound
+        // to be larger
+        private float? _lowerWarningPercentage;
+        [ViewVariables]
+        [DataField("lowerWarnAround")]
+        public float? LowerWarningPercentage
+        {
+            get => _lowerWarningPercentage;
+            set
+            {
+                if (value == null)
+                {
+                    _lowerWarningPercentage = null;
+                    return;
+                }
+
+                var testValue = value * LowerBound;
+                if (value < 1f
+                    || testValue > UpperWarningBound
+                    || testValue > UpperBound)
+                    return;
+
+                _lowerWarningPercentage = value;
+            }
+        }
+
+        [ViewVariables]
+        public float? UpperWarningBound {
+            get
+            {
+                if (UpperBound == null || UpperWarningPercentage == null) return null;
+
+                return UpperBound * UpperWarningPercentage;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    UpperWarningPercentage = null;
+                    return;
+                }
+
+                if (UpperBound != null)
+                {
+                    float? percentage = value / UpperBound;
+                    if (percentage != null)
+                        UpperWarningPercentage = percentage;
+                }
+            }
+        }
+
+        [ViewVariables]
+        public float? LowerWarningBound
+        {
+            get
+            {
+                if (LowerBound == null || LowerWarningPercentage == null) return null;
+
+                return LowerBound * LowerWarningPercentage;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    LowerWarningPercentage = null;
+                    return;
+                }
+
+                if (LowerBound != null)
+                {
+                    float? percentage = value / LowerBound;
+                    if (percentage != null)
+                        LowerWarningPercentage = percentage;
+                }
+            }
+        }
```

Comment:
extraordinarily ugly duplication

Reply:
this is one of the bigger ones i'll focus on since this is one of the core pieces of how the air alarm works - if the getter/setter thing is too ugly then i'll just find a compact way of doing it i suppose